File: split/mathac.pas
procedure mathac;
begin
  if curcmd = 45 then{1166:}

  begin
    begin
      if interaction = 3 then;
      printnl(262);
      print(1138);
    end;
    printesc(523);
    print(1139);
    begin
      helpptr := 2;
      helpline[1] := 1140;
      helpline[0] := 1141;
    end;
    error;
  end{:1166}
;
  begin
    mem[curlist.tailfield].hh.rh := getnode(5);
    curlist.tailfield := mem[curlist.tailfield].hh.rh;
  end;
  mem[curlist.tailfield].hh.b0 := 28;
  mem[curlist.tailfield].hh.b1 := 0;
  mem[curlist.tailfield + 1].hh := emptyfield;
  mem[curlist.tailfield + 3].hh := emptyfield;
  mem[curlist.tailfield + 2].hh := emptyfield;
  mem[curlist.tailfield + 4].hh.rh := 1;
  scanfifteenbitint;
  mem[curlist.tailfield + 4].hh.b1 := curval mod 256 + 0;
  if (curval >= 28672) and ((eqtb[5307].int >= 0) and (eqtb[5307].int < 16)) then
    mem[
      curlist.tailfield + 4].hh.b0 := eqtb[5307].int
  else
    mem[curlist.tailfield + 4].hh.b0 := (curval div 256) mod 16;
  scanmath(curlist.tailfield + 1);
end;

File: cpp/mathac.cpp
#include "mathac.h"
#include "printnl.h"
#include "print.h"
#include "printesc.h"
#include "error.h"
#include "getnode.h"
#include "scanfifteenbitint.h"
#include "scanmath.h"

void mathac(void)
{
	if (curcmd == accent)
	{
		if (interaction == 3)
			printnl(262); //! 
		print(1138); //Please use 
		printesc(523); //mathaccent
		print(1139); // for accents in math mode
		helpptr = 2;
		helpline[1] = 1140; //I'm changing \accent to \mathaccent here; wish me luck.
		helpline[0] = 1141; //(Accents are not the same in formulas as they are in text.)
		error();
	}
	tail_append(getnode(5));
	type(tail) = 28;
	subtype(tail) = 0;
	mem[tail+1].hh = emptyfield;
	mem[tail+3].hh = emptyfield;
	mem[tail+2].hh = emptyfield;
	link(tail+4) = 1;
	scanfifteenbitint();
	subtype(tail+4) = curval%0x1'00;
	if (curval >= 0x7000 && int_par(cur_fam_code) >= 0 && int_par(cur_fam_code) < 0x10)
		type(tail+4) = int_par(cur_fam_code);
	else
		type(tail+4) = (curval/0x1'00)%0x10;
	scanmath(tail+1);
}
